packages/components/src/utils/src/types.ts

```diff
@@ -42,6 +42,11 @@ export interface BaseComponentProps extends DOMProps, AriaLabelingProps, SlotPro
 
 }
 
+// TODO: For a incremental transition, we can start with this
+export interface RenderComponentProps<T> extends RenderProps<T>, AriaLabelingProps, SlotProps {
```

Comment:
I'm not sure it makes sense to call this all RenderComponentProps since AriaLabelingProps, SlotProps wouldn't be considered render props

Reply:
It's temporary, I'll switch it all to BaseComponentProps later. Should I rename for clarity?

Reply:
ya might as well rename now.